File: app/java/src/main/java/com/example/production/domain/model/bom/BomExplosion.java
package com.example.production.domain.model.bom;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BomExplosion {
    private String parentItemCode;
    private String childItemCode;
    private LocalDate effectiveFrom;
    private LocalDate effectiveTo;
    private BigDecimal baseQuantity;
    private BigDecimal requiredQuantity;
    private BigDecimal defectRate;
    private Integer sequence;
    private Integer level;
    private BigDecimal totalQuantity;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Bom/BomExplosion.cs
namespace ProductionManagement.Domain.Models.Bom;

/// <summary>
/// BOM 展開結果
/// </summary>
public class BomExplosion
{
    public required string ParentItemCode { get; set; }

    public required string ChildItemCode { get; set; }

    public DateOnly EffectiveFrom { get; set; }

    public DateOnly? EffectiveTo { get; set; }

    public decimal BaseQuantity { get; set; }

    public decimal RequiredQuantity { get; set; }

    public decimal DefectRate { get; set; }

    public int? Sequence { get; set; }

    public int Level { get; set; }

    public decimal TotalQuantity { get; set; }
}
